dbms/src/Storages/MergeTree/DataPartsExchange.cpp

```diff
@@ -200,7 +200,9 @@ MergeTreeData::MutableDataPartPtr Fetcher::fetchPart(
     String tmp_prefix = tmp_prefix_.empty() ? TMP_PREFIX : tmp_prefix_;
 
     String relative_part_path = String(to_detached ? "detached/" : "") + tmp_prefix + part_name;
-    String absolute_part_path = data.getFullPath() + relative_part_path + "/";
+    auto reservation = data.reserveSpaceForPart(0); ///@TODO_IGR ASK What size should be there?
```

Comment:
I think we have to add new http handler (or params to existing handler) which will tell about part size.